zipline-loader/src/main/kotlin/app/cash/zipline/loader/ZiplineLoader.kt

```diff
@@ -22,20 +37,24 @@ class ZiplineLoader(
 //  val cacheMaxSizeInBytes: Int = 100 * 1024 * 1024,
   val client: ZiplineHttpClient,
 ) {
+  suspend fun load(scope: CoroutineScope, zipline: Zipline, url: String) {
+    //  TODO load manifest from URL then call load(...manifest)
+  }
+
   suspend fun load(scope: CoroutineScope, zipline: Zipline, manifest: ZiplineManifest) {
     // TODO consider moving this to `async` on the Zipline's dispatcher, which is guaranteed single-threaded
     val ziplineMutex = Mutex()
     val concurrentDownloadsSemaphore = Semaphore(3)
 
-    val loads = manifest.files.map {
-      ModuleLoad(zipline, ziplineMutex, concurrentDownloadsSemaphore, it, mutableListOf())
+    val loads = manifest.modules.map {
+      ModuleLoad(zipline, ziplineMutex, concurrentDownloadsSemaphore, it.key, it.value, mutableListOf())
     }
 
-    val idToLoad = loads.associateBy { it.module.id }
+    val idToLoad = loads.associateBy { it.id }
 
     val loadsSorted = loads.topologicalSort { load ->
       load.module.dependsOnIds.map { upstream ->
-        idToLoad[upstream] ?: throw IllegalArgumentException("${load.module.id} depends on unknown module $upstream")
+        idToLoad[upstream] ?: throw IllegalArgumentException("${load.id} depends on unknown module $upstream")
```

Comment:
skip the topological sort now that the manifest does it?

(might want to move this nice exception to the manifest instead of `!!`)